core/corehttp/gateway_handler.go

```diff
@@ -82,6 +82,11 @@ func (i *gatewayHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if r.Method == "OPTIONS" {
```

Comment:
I feel we should always produce `OPTIONS` response.
Gateway owner can decide what types of requests are accepted by customizing  `Access-Control-Allow-Method` header (`GET` is ro, `POST`, `PUT`, `DELETE` are rw). 
All major browsers respect this header and refuse to do XHR requests if method is not whitelisted.